pkg/api/graph/types.go

```diff
@@ -2,603 +2,25 @@ package graph
-	return n.ImageStream
-}
-
-func (n ImageStreamNode) String() string {
-	return fmt.Sprintf("<ImageStream %s/%s>", n.Namespace, n.Name)
-}
-
-func (*ImageStreamNode) Kind() int {
-	return ImageStreamGraphKind
-}
-
-func imageStreamName(stream *image.ImageStream) UniqueName {
-	return UniqueName(fmt.Sprintf("%d|%s", ImageStreamGraphKind, stream.Status.DockerImageRepository))
-}
-
-// ImageStream adds a graph node for the Image Stream if it does not already exist.
-func ImageStream(g MutableUniqueGraph, stream *image.ImageStream) graph.Node {
-	return EnsureUnique(g,
-		imageStreamName(stream),
-		func(node Node) graph.Node {
-			return &ImageStreamNode{node, stream}
-		},
-	)
-}
-
-func FindImageStream(g MutableUniqueGraph, stream *image.ImageStream) graph.Node {
-	return g.Find(imageStreamName(stream))
-}
-
-type ReplicationControllerNode struct {
-	Node
-	*kapi.ReplicationController
-}
-
-func (n ReplicationControllerNode) Object() interface{} {
-	return n.ReplicationController
-}
-
-func (n ReplicationControllerNode) String() string {
-	return fmt.Sprintf("<replication controller %s/%s>", n.Namespace, n.Name)
-}
-
-func (*ReplicationControllerNode) Kind() int {
-	return ReplicationControllerGraphKind
-}
-
-// ReplicationController adds a graph node for the ReplicationController if it does not already exist.
-func ReplicationController(g MutableUniqueGraph, rc *kapi.ReplicationController) graph.Node {
-	return EnsureUnique(g,
-		UniqueName(fmt.Sprintf("%d|%s/%s", ReplicationControllerGraphKind, rc.Namespace, rc.Name)),
-		func(node Node) graph.Node {
-			return &ReplicationControllerNode{node, rc}
-		},
-	)
-}
-
-type ImageLayerNode struct {
-	Node
-	Layer string
-}
-
-func (n ImageLayerNode) Object() interface{} {
-	return n.Layer
-}
-
-func (n ImageLayerNode) String() string {
-	return fmt.Sprintf("<image layer %s>", n.Layer)
-}
-
-func (*ImageLayerNode) Kind() int {
-	return ImageLayerGraphKind
-}
-
-// ImageLayer adds a graph node for the layer if it does not already exist.
-func ImageLayer(g MutableUniqueGraph, layer string) graph.Node {
-	return EnsureUnique(g,
-		UniqueName(fmt.Sprintf("%d|%s", ImageLayerGraphKind, layer)),
-		func(node Node) graph.Node {
-			return &ImageLayerNode{node, layer}
-		},
-	)
-}
-
-type BuildNode struct {
-	Node
-	Build *build.Build
-}
-
-func (n BuildNode) Object() interface{} {
-	return n.Build
-}
-
-func (n BuildNode) String() string {
-	return fmt.Sprintf("<build %s/%s>", n.Build.Namespace, n.Build.Name)
-}
-
-func (*BuildNode) Kind() int {
-	return BuildGraphKind
-}
 
-// Build adds a graph node for the build if it does not already exist.
-func Build(g MutableUniqueGraph, build *build.Build) graph.Node {
-	return EnsureUnique(g,
-		UniqueName(fmt.Sprintf("%d|%s/%s", BuildGraphKind, build.Namespace, build.Name)),
-		func(node Node) graph.Node {
-			return &BuildNode{node, build}
-		},
-	)
+	return UniqueName(fmt.Sprintf("%d|%s/%s", nodeKind, meta.Namespace, meta.Name))
```

Comment:
I'm guessing the `%d` that corresponds to `nodeKind` should now be `%s` instead?